python312Packages.pysmi: 0.3.4 -> 1.4.4; python312Packages.pysnmp: 4.4.2 -> 6.2.2

mweinelt opened the pull request:
Removes the lextudio forks, which now have become the canonical pysmi/pysnmp source.

https://github.com/lextudio/pysmi/compare/refs/tags/v0.3.4...v1.4.4
https://github.com/lextudio/pysnmp/blob/v6.2.2/CHANGES.rst

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt committed:
python312Packages.pysmi: 0.3.4 -> 1.4.4

https://github.com/lextudio/pysmi/blob/v1.4.4/CHANGES.rst

Absorbs pysmi-lextudio, since the author of pysmi has passed away and
the lextudio fork has now become pysmi.

mweinelt commented:
Result of `nixpkgs-review pr 328047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>blocksat-cli (python312Packages.blocksat-cli)</li>
    <li>blocksat-cli.dist (python312Packages.blocksat-cli.dist)</li>
    <li>home-assistant-component-tests.snmp</li>
    <li>routersploit</li>
    <li>routersploit.dist</li>
  </ul>
</details>
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>glances</li>
    <li>glances.dist</li>
    <li>home-assistant-component-tests.brother</li>
    <li>python311Packages.atenpdu</li>
    <li>python311Packages.atenpdu.dist</li>
    <li>python311Packages.blocksat-cli</li>
    <li>python311Packages.blocksat-cli.dist</li>
    <li>python311Packages.brother</li>
    <li>python311Packages.brother.dist</li>
    <li>python311Packages.patator</li>
    <li>python311Packages.patator.dist</li>
    <li>python311Packages.pysmi</li>
    <li>python311Packages.pysmi.dist</li>
    <li>python311Packages.pysnmp</li>
    <li>python311Packages.pysnmp.dist</li>
    <li>python312Packages.atenpdu</li>
    <li>python312Packages.atenpdu.dist</li>
    <li>python312Packages.brother</li>
    <li>python312Packages.brother.dist</li>
    <li>python312Packages.patator</li>
    <li>python312Packages.patator.dist</li>
    <li>python312Packages.pysmi</li>
    <li>python312Packages.pysmi.dist</li>
    <li>python312Packages.pysnmp</li>
    <li>python312Packages.pysnmp.dist</li>
  </ul>
</details>

mweinelt commented:
Routersploit is blocked by the removal of the sync high-level API.

https://github.com/threat9/routersploit/issues/819

dotlambda commented on pkgs/development/python-modules/pysnmp/default.nix:
```diff
@@ -1,40 +1,74 @@
   pname = "pysnmp";
-  version = "4.4.12";
-  format = "setuptools";
+  version = "6.2.2";
```
```suggestion
  version = "6.2.5";
```

dotlambda commented on pkgs/servers/home-assistant/update-component-packages.py:
```diff
@@ -44,6 +44,7 @@
     "numpy": "numpy_1",
     "ollama-hass": "ollama",
     "paho-mqtt": "paho-mqtt_1",
+    "pysnmp-lextudio": "pysnmp",
```
Not needed anymore: https://github.com/NixOS/nixpkgs/pull/338722

dotlambda commented on pkgs/servers/home-assistant/default.nix:
```diff
@@ -540,6 +540,8 @@ in python.pkgs.buildPythonApplication rec {
       ffmpeg = "${lib.getBin ffmpeg-headless}/bin/ffmpeg";
     })
+
+    ./patches/pysnmp-compat.patch
```
also no longer needed

mweinelt committed:
python312Packages.pysnmp: 4.4.2 -> 6.2.5

https://github.com/lextudio/pysnmp/blob/v6.2.5/CHANGES.rst
https://github.com/etingof/pysnmp/issues/429

Promotes the lextudio fork to the the primary pysnmp package.
